modules/optimization/src/optimizationreporters/OptimizationReporterBase.C

```diff
@@ -71,15 +81,22 @@ OptimizationReporterBase::setSimulationValuesForTesting(std::vector<Real> & data
 void
 OptimizationReporterBase::computeGradient(libMesh::PetscVector<Number> & gradient) const
 {
-  for (const auto & p : make_range(_nparams))
-    if (_gradients[p]->size() != _nvalues[p])
+  for (const auto & param_group_id : make_range(_nparams))
+  {
+    if (_gradients[param_group_id]->size() != _nvalues[param_group_id])
       mooseError("The gradient for parameter ",
-                 _parameter_names[p],
+                 _parameter_names[param_group_id],
                  " has changed, expected ",
-                 _nvalues[p],
+                 _nvalues[param_group_id],
                  " versus ",
-                 _gradients[p]->size(),
+                 _gradients[param_group_id]->size(),
                  ".");
+    auto params = _parameters[param_group_id];
+    auto grads = _gradients[param_group_id];
+    for (const auto & param_id : make_range(_nvalues[param_group_id]))
+      grads->at(param_id) += params->at(param_id) * _tikhonov_coeff;
```

Comment:
```suggestion
      (*grads)[param_id] += (*params)[param_id] * _tikhonov_coeff;
```

Reply:
Why? Please don't do this - bracket operator can have accidental insertion. You expect the param ID to be in the map, so write code that describes that intention (at has bounds checking).

Reply:
...I say this after spending multiple days debugging... not realizing that an accidental insertion was being done silently without any error

Reply:
I keep forgetting about this.

Reply:
These aren't maps, they are vectors...

Reply:
Bracket is more appropriate, it's not inserting anything.

Reply:
Looks good to me. map::operator[] triggers me